src/Data/ConfessionData.re

```diff
@@ -0,0 +1,103 @@
+type confessionComment = {
+  message: string,
+  id: string,
+  ts: int,
+};
+
+type comments = {commentsArray: array(confessionComment)};
+
+type confession = {
+  message: string,
+  id: string,
+  ts: int,
+  comments,
+};
+
+type allConfessions = {allConfessionsArray: array(confession)};
+
+type data = {allConfessions};
+
+type allConfessionsResponse = {data};
+
+type confessionWithCommentsdata = {confession};
+
+type confessionWithCommentsResponse = {confessionWithCommentsdata};
+
+type createCommentResponse = {confessionWithCommentsResponse};
+
+type recentConfessions = array(confession);
+
+module Decode = {
```

Comment:
Again organizing these by function "decode" seems weird to me compared to organizing by type (ex. Comment).

It would make more sense to me to have a `Comment` file with the type def and the decode function all wrapped together.

Reply:
these things are more likely to change together then all of the things you need to decode, that's why it's better to group them that way